src/index.js

```diff
@@ -333,14 +333,21 @@ class HCaptcha extends React.Component {
 
 
         if (!this.isReady()) {
-            return;
+            if(opts?.async){
+              return queueMicrotask(() => hcaptcha.execute(captchaId, opts))
```

Comment:
@gabrielferreira-imi Are we sure this functionality (queueMicrotask) is safe to use in all browsers?

Reply:
Maybe something like this can handle any queued execution callback and potentially handle async vs using callbacks?

```jsx
class HCaptcha {
 
  constructor() {
    this._onReady = null;
  }

  renderCaptcha(onReady) {
    this.setState({ isRemoved: false, captchaId }, () => {
      onReady && onReady();
      this._onReady && this._onReady(id);
    });
  }

  execute(opts) {

    if (!this.isReady()) {
      const onReady = new Promise((resolve, reject) => {
        this._onReady = (id) => {
          try {
            if (!opts.async) {
              resolve();
              hcaptcha.execute(id, opts);
            } else {
               hcaptcha.execute(id, opts)
                 .then(resolve)
                 .catch(reject)
            }
          } catch(e) {
             reject(e);
          }
        }
    });
   
    return opts?.async ?  onReady : null;
  }
  
  return hcaptcha.execute(id, opts);
}

```

Reply:
@brdlyptrs I don't know how far we need to provide support, but it's widely accepted
https://developer.mozilla.org/en-US/docs/Web/API/queueMicrotask#browser_compatibility
![CleanShot 2024-08-08 at 21 18 26@2x](https://github.com/user-attachments/assets/b7d2dcb2-b67d-4002-826e-d38fe62be45c)

Reply:
I suggest that we do something like this (tested locally, seems to be working):

```
const MAX_EXECUTION_TRIALS = 3;

constructor (props) {
  super(props);

  this.executionCounts = 0;
  // ...
}

handleExecuteTrial (opts = null) {
  this.executionCounts += 1;

  if (this.executionCounts > MAX_EXECUTION_TRIALS) {
    if (opts && opts.async) {
      return Promise.resolve({});
    }

    return;
  }

  if (opts && opts.async) {
    return new Promise(resolve => setTimeout(() => {
      resolve(this.execute(opts));
    }, 500));
  } else {
    setTimeout(() => {
      this.execute(opts);
    }, 500);
  }
}

execute (opts = null) {
  try {
    const { captchaId } = this.state;
    const hcaptcha = this._hcaptcha;

    if (!this.isReady()) {
        return this.handleExecuteTrial(opts);
    }

    this.executionCounts = 0;

    if (opts && typeof opts !== "object") {
        opts = null;
    }

    return hcaptcha.execute(captchaId, opts);
  } catch (error) {
      this.sentryHub.captureException(error);
  }
}
```

Reply:
[Draft just for visibility](https://github.com/hCaptcha/react-hcaptcha/pull/239/files)

Reply:
@zoryana94 I'd avoid anything with a timer, that sets up for points of failure if the timers are not cleared or handled correctly. 

Why do you believe this is better than storing the reference to the call itself? You can only execute once per hCaptcha instance, we should gate on that concept.

Reply:
I just went for the quick and obvious working way
from what I know, clearing the timer in this case should be not much of a problem, additionally, we could clear it at the start in case if the function is called multiple times (for any reason)
but it was only a suggestion, feel free to go with any other way of implementation, @gabrielferreira-imi